src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/NameSyntaxClassifier.cs

```diff
@@ -92,6 +92,31 @@ CandidateReason.Ambiguous or
                 return true;
             }
 
+            if (name is IdentifierNameSyntax { Identifier.Text: "async" } && symbol is null)
+            {
+                // MemberDeclarationSyntax examle:
+                //
+                // class Test
+                // {
+                //   public async
+                // }
+                //
+                // ExpressionStatementSyntax example:
+                //
+                // void M()
+                // {
+                //   async
+                // }
+                //
+                // User is probably defining an async local function here
+                if ((name.Parent is IncompleteMemberSyntax incompletemember && !incompletemember.Modifiers.Any(SyntaxKind.AsyncKeyword)) ||
+                    name.Parent is ExpressionStatementSyntax)
```

Comment:
do we have tests for things like a partially written lambda as an expression?

Reply:
Added cases like `Action a = async `. Is this what you had in your head?